### PR TITLE
Site Settings: Add more settings Composing card for Jetpack sites

### PR DESCRIPTION
**Pull request by tyxla**
This PR implements After the Deadline settings for Jetpack sites in the Writing > Composing card. It also moves the current Markdown and Default Post Format settings in separate components in order to keep the main Composing card component clean. It also introduces a `setFieldValue` method to `wrapSettingsForm` that allows manually setting a specific field's value. The PR is part of #9171.

Preview (with collapsed Advanced settings):

Preview (with expanded Advanced settings):

To test:

* Checkout this branch
* Go to `/settings/writing/$site` where `$site` is one of your Jetpack sites.
* Verify the default post format field works properly like before.
* Verify the Markdown field is not displayed.
* Play with the new After the Deadline settings (main toggle and sub settings) and verify they are stored and retrieved properly.
* Go to `/settings/writing/$site` where `$site` is one of your WordPress.com sites.
* Verify the default post format field works properly like before.
* Verify the Markdown field works properly like before.
* Verify the After the Deadline settings are not displayed for this site.

Styles here could benefit from #11240 - I can update the PR once it's merged.

### CODE REVIEW

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+						<ExternalLink href={ 'https://jetpack.com/support/spelling-and-grammar/' } target="_blank">
+							{ translate( 'Learn more about After the Deadline' ) }
+						</ExternalLink>
+					</InfoPopover>
+				</div>
+
+				<JetpackModuleToggle
+					siteId={ selectedSiteId }
+					moduleSlug="after-the-deadline"
+					label={ translate( 'Check your spelling, style, and grammar.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Spelling and Grammar' )` *ES Score: 8.14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	renderEnglishOptionsSection() {
+		const { afterTheDeadlineModuleActive, translate } = this.props;
+
+		return (
+			<div className="composing__module-settings is-indented">
+				<FormLegend>
+					{ translate( 'English Options' ) }
+				</FormLegend>
+				<FormSettingExplanation>
+					{ translate( 'Enable proofreading for the following grammar and style rules:' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **40** times:
`translate( 'Enable proofreading for the following grammar and style rules when writing posts and pages:' )` *ES Score: 15.58*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	renderAutoLanguageDetectionSection() {
+		const { afterTheDeadlineModuleActive, translate } = this.props;
+
+		return (
+			<div className="composing__module-settings is-indented">
+				<FormLegend>
+					{ translate( 'Automatic Language Detection' ) }
+				</FormLegend>
+				<FormSettingExplanation>
+					{ translate( 'The proofreader supports English, French, German, Portuguese and Spanish.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'The proofreader supports English, French, German, Portuguese, and Spanish. Your <a href="options-general.php">blog language</a> setting is the default proofreading language.' )` *ES Score: 11.38*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+				{
+					this.renderToggle( 'onpublish', ! afterTheDeadlineModuleActive, translate(
+						'A post or page is first published'
+					) )
+				}
+
+				{
+					this.renderToggle( 'onupdate', ! afterTheDeadlineModuleActive, translate(
+						'A post or page is updated'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **47** times:
`translate( 'a post or page is updated' )` *ES Score: 12.39*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+				<FormLegend>
+					{ translate( 'Proofreading' ) }
+				</FormLegend>
+				<FormSettingExplanation>
+					{ translate( 'Automatically proofread content when:' ) }
+				</FormSettingExplanation>
+
+				{
+					this.renderToggle( 'onpublish', ! afterTheDeadlineModuleActive, translate(
+						'A post or page is first published'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **45** times:
`translate( 'a post or page is first published' )` *ES Score: 13.52*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
This looks good. I imagine the other PR to standardize the right-margin of toggles will take care of the misalignment here.

The only other thing I noticed was the lack of initial capitals on the first two proofreading settings:

screen shot 2017-02-08 at 10 21 38 am

All our other labels have an initial capital letter, we should standardize.

**Comment by rickybanister** [comment]
To follow up, if we don't like that those two labels start with `a...` we could change them to be plural:

`Posts or pages are first published`
`Posts or pages are updated`

**Comment by tyxla** [comment]
> This looks good. I imagine the other PR to standardize the right-margin of toggles will take care of the misalignment here.

Yes - I'll make sure to do any necessary updates/cleanups here once that PR lands.

> To follow up, if we don't like that those two labels start with a... we could change them to be plural:

Even better! I've updated the labels to the one you suggested. 

@rickybanister do you feel there's anything else we should improve, design-wise? Also cc @MichaelArestad for a second opinion.

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+				{
+					this.renderToggle( 'onpublish', ! afterTheDeadlineModuleActive, translate(
+						'Posts or pages are first published'
+					) )
+				}
+
+				{
+					this.renderToggle( 'onupdate', ! afterTheDeadlineModuleActive, translate(
+						'Posts or pages are updated'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **47** times:
`translate( 'a post or page is updated' )` *ES Score: 7.10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+				<FormLegend>
+					{ translate( 'Proofreading' ) }
+				</FormLegend>
+				<FormSettingExplanation>
+					{ translate( 'Automatically proofread content when:' ) }
+				</FormSettingExplanation>
+
+				{
+					this.renderToggle( 'onpublish', ! afterTheDeadlineModuleActive, translate(
+						'Posts or pages are first published'
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **45** times:
`translate( 'a post or page is first published' )` *ES Score: 9.02*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by MichaelArestad** [comment]
@tyxla We should hide the majority of those settings in an "Advanced options" section like we are doing in the `feature/settings-overhaul` branch of Jetpack:

image

**Comment by tyxla** [comment]
@MichaelArestad of course! Thanks for pointing that out. It's now implemented in 202a41a9f61c07a90e527d0bcdd2a5afc36c28f2, I've also updated the previews in the PR description. Feel free to have another 👁 .

File: client/my-sites/site-settings/composing/style.scss [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.composing__module-settings.is-indented {
+	margin: 16px 32px 0;
```
Comment (MichaelArestad) [inline client/my-sites/site-settings/composing/style.scss]: Can we drop this in favor of the solution offered here: https://github.com/Automattic/wp-calypso/pull/11240

**Comment by MichaelArestad** [comment]
Visually, looks :100:

Nice work.

File: client/my-sites/site-settings/composing/style.scss (hunk 8 above)
Comment (tyxla) [inline client/my-sites/site-settings/composing/style.scss]: Yes, thanks for pointing that out. I've already added that to the description of the PR, and I'm planning to address it after it's merged.

**Comment by oskosk** [comment]
Is it possible that some ATD settings still need to be added to the whitelist in `state/jetpack/settings/utils.js`?

**Comment by tyxla** [comment]
> Is it possible that some ATD settings still need to be added to the whitelist in state/jetpack/settings/utils.js?

Of course! I totally missed that. Thanks @oskosk, good catch! Added in 65144ec948d730e7b0aa4a283feb3775ac1f792b.

File: client/my-sites/site-settings/composing/style.scss (hunk 8 above)
Comment (tyxla) [inline client/my-sites/site-settings/composing/style.scss]: @MichaelArestad this has been implemented in 76aa0071685542ed57d5f37cee65fcde051647f2.

**Comment by tyxla** [comment]
This could use a new review @oskosk 😉

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+			selectedSiteId,
+			translate
+		} = this.props;
+
+		return (
+			<FormFieldset>
+				<div className="composing__info-link-container">
+					<InfoPopover position={ 'left' }>
+						<ExternalLink href={ 'https://jetpack.com/support/spelling-and-grammar/' } target="_blank">
+							{ translate( 'Learn more about After the Deadline' ) }
```
Comment (oskosk) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: I'm not quite sure if this should read `Learn more about After the Deadline` or `Learn more about Spelling and Grammar`. Although the latter would sound like we're trying to school the user: P. This could be the topic for another PR though.

File: client/my-sites/site-settings/composing/index.jsx [hunk 10]
```diff
@@ -90,13 +70,22 @@ Composing.defaultProps = {
...
 };
 
-export default localize( Composing );
+export default connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+
+		return {
+			jetpackSettingsUISupported: siteSupportsJetpackSettingsUi( state, siteId ),
+			siteIsJetpack: isJetpackSite( state, siteId ),
```
Comment (oskosk) [inline client/my-sites/site-settings/composing/index.jsx]: Why backwards? Weren't we using  `props.isJetpackSite` at some other places ?. Well I've just realized out why :P

File: client/my-sites/site-settings/composing/after-the-deadline.jsx (hunk 9 above)
Comment (tyxla) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: 😆 

Well, the module label is "Spelling and Grammer" in Jetpack indeed, so what you say makes sense. In the same time, I kinda agree with you on the teaching the user argument. Perhaps it's best if we get some more opinions on this? @MichaelArestad @rickybanister what do you think about this one?

**Comment by oskosk** [comment]
LGTM. Every settings works as expected. The comment I left about the text in the InfoPopover can be handled later. 🚢

**Comment by rickybanister** [comment]
Two very minor things we noticed:

It's been decided to remove all the punctuation from labels, so no need for the `.` at the end of the `Check your spelling, style, and grammar`

And last thing—can 'Advanced Options' be aligned with the text label above since those options are children of the parent toggle. Only when the advanced options are related to the section _in general_ would we want it to be fully left-aligned.

**Comment by tyxla** [comment]
@rickybanister minor things addressed. A small preview of the updated version: 


Feel free to have another look.

File: client/my-sites/site-settings/composing/after-the-deadline.jsx [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+						</ExternalLink>
+					</InfoPopover>
+				</div>
+
+				<JetpackModuleToggle
+					siteId={ selectedSiteId }
+					moduleSlug="after-the-deadline"
+					label={ translate( 'Check your spelling, style, and grammar' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/after-the-deadline.jsx]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Spelling and Grammar' )` *ES Score: 8.15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
@tyxla beautiful, ship it.

**Comment by folletto** [comment]
Sorry if I missed this before, but wouldn't be better here to use our foldable card pattern instead of a link to open "Advanced settings"?

Something like this:

foldable-spelling-card

**Comment by MichaelArestad** [comment]
@folletto That's been brought up recently by @rickybanister as well. It's something we're definitely exploring, but this works and is good to merge for now.
